core/src/main/java/feign/SynchronousMethodHandler.java

```diff
@@ -43,13 +43,14 @@
   private final ErrorDecoder errorDecoder;
   private final boolean decode404;
   private final boolean closeAfterDecode;
+  private final boolean propagateExceptions;
```

Comment:
I like the PR and the general idea....

The only think I would ask is to replace this `boolean propagateExceptions` with an enum `ExceptionPropagationPolicy propagationPolicy`

Loved the `builder. propagateExceptions()` idea

Reply:
Sounds good to me, updates soon.